### PR TITLE
[3.4] bpo-34623: Use XML_SetHashSalt in _elementtree

### PR DESCRIPTION
**Pull request by stratakis**
Backport for the 3.4 branch.

https://bugs.python.org/issue34623

### CODE REVIEW

File: Modules/_elementtree.c [hunk 1]
```diff
@@ -3259,6 +3259,11 @@ xmlparser_init(PyObject *self, PyObject *args, PyObject *kwds)
         PyErr_NoMemory();
         return -1;
     }
+    /* expat < 2.1.0 has no XML_SetHashSalt() */
+    if (EXPAT(SetHashSalt) != NULL) {
+        EXPAT(SetHashSalt)(self_xp->parser,
```
Comment (stratakis) [inline Modules/_elementtree.c:3264]: This is the only line that has been modified from the previous PR's. Namely it's `self_xp->parser `on 3.4. On the other branches it's `self->parser` due to argumentclinication.

**Review by vstinner — APPROVED** [review]
LGTM.

@tiran: Would you mind to review it as well?

**Comment by vstinner** [comment]
Oh, Travis CI failed on a random failure:

```
======================================================================
ERROR: test_ignore (test.test_multiprocessing_forkserver.TestIgnoreEINTR)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/python/cpython/Lib/test/_test_multiprocessing.py", line 3623, in test_ignore
    os.kill(p.pid, signal.SIGUSR1)
ProcessLookupError: [Errno 3] No such process
```

I scheduled a new job.

**Comment by vstinner** [comment]
I closed/reopened the PR to trigger a new Travis CI job.

**Comment by vstinner** [comment]
@larryhastings: Hi Larry, would you mind to merge this security fix?

**Comment by larryhastings** [comment]
Thank for the backported fix!

**Pull request merged**